### PR TITLE
Fix site on dropdown when its deleted

### LINKED ISSUE #327: User dropdown doesn't get updated after site deletion

### PR DESCRIPTION
**Pull request by ArsalanSavand**
Fixes #327

### CODE REVIEW

File: app/site/site_controller.js [hunk 1]
```diff
@@ -100,7 +100,8 @@ function SiteController($scope, $rootScope, $state, $stateParams, $mdToast,
 
         $rootScope.$broadcast("gonevisDash.SiteController:remove");
         $mdToast.showSimple("Site deleted");
-        $state.go($scope.user.sites ? "site-new" : "dash.main");
+        // An error while no sites are available, FIX THIS LINE.
+        $state.go($scope.user.sites ? "dash.main" : "site-new");
```
Comment (Alir3z4) [inline app/site/site_controller.js]: @amirm3hdi what do you think of this ?

File: app/header/header_controller.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ function HeaderController($scope, $rootScope, $state, $stateParams, $mdToast, Au
     $state.go("signin");
   });
 
-  $scope.$on("gonevisDash.SiteSettingsController:remove", function () {
+  $scope.$on("gonevisDash.SiteController:remove", function () {
```
Comment (amirm3hdi) [inline app/header/header_controller.js]: @ArsalanSavand Are you sure you renamed it everywhere? like where it's called?

File: app/site/site_controller.js (hunk 1 above)
Comment (amirm3hdi) [inline app/site/site_controller.js]: @Alir3z4 I'll get on this later, I need to fix and update it myself.

File: app/header/header_controller.js (hunk 2 above)
Comment (Alir3z4) [inline app/header/header_controller.js]: We need to write more tests so it covers the situations like this.

**Comment by Alir3z4** [comment]
@amirm3hdi @ArsalanSavand make this ready for merge please/

**Comment by amirm3hdi** [comment]
@Alir3z4 All done.

File: app/site/site_controller.js (hunk 1 above)
Comment (amirm3hdi) [inline app/site/site_controller.js]: Done

**Pull request merged**